lib/init/grass.py

```diff
@@ -1901,6 +1901,94 @@ def csh_startup(location, location_name, mapset, grass_env_file):
     return process
 
 
+def zsh_startup(location, location_name, grass_env_file):
```

Comment:
I would recommend to add a new boolean flag `zsh` (`False` by default) to `bash_startup()` instead of duplicating most of the code from the latter function. It will help us maintain the code later.

Reply:
I choose to rename to `shell_startup(location, location_name, grass_env_file, shell_name)` and pass to `shell_name` either bash or zsh.
Semantically, it felt awkward with a zsh flag to a bash function.